tests/optimizers/test_dion_numerical.py

```diff
@@ -0,0 +1,133 @@
+import pytest
+import torch
+import numpy as np
+from typing import Tuple
+import math
+
+from optimizers.dion_reference import (
+    dion_update, power_iteration, orthogonalize,
+    fix_all_zero_or_nan
+)
+
+
+class TestDionNumericalAccuracy:
+    """Test numerical accuracy and stability of Dion optimizer components"""
+    
+    @pytest.fixture
+    def device(self):
+        return torch.device("cuda" if torch.cuda.is_available() else "cpu")
+    
+    def test_orthogonalization_stability(self, device):
+        """Test numerical stability of orthogonalization methods"""
+        torch.manual_seed(42)
+        
+        # Test with ill-conditioned matrices
+        n = 50
+        # Create matrix with large condition number
+        U, S, Vt = torch.linalg.svd(torch.randn(n, n, device=device))
+        S_modified = torch.logspace(0, -10, n, device=device)  # Condition number ~1e10
+        A = U @ torch.diag(S_modified) @ Vt
+        
+        # Test different QR methods
+        methods = ["qr", "cqr", "rcqr"]
+        for method in methods:
+            try:
+                rng = torch.Generator(device=device)
+                rng.manual_seed(42)
+                Q = orthogonalize(A, qr_method=method, rng=rng)
+                
+                # Check orthogonality (within reasonable tolerance for ill-conditioned matrices)
+                if Q.shape[0] >= Q.shape[1]:
+                    QtQ = Q.T @ Q
+                    I = torch.eye(Q.shape[1], device=device, dtype=Q.dtype)
+                    ortho_error = torch.max(torch.abs(QtQ - I)).item()
+                    assert ortho_error < 1e-3, f"Method {method}: orthogonality error {ortho_error}"
+                    
+            except Exception as e:
```

Comment:
Change this to only catch exception for method "cqr". Using "qr" or "rcqr" should never fail due to poorly conditioned matrices.